packages/flutter_tools/lib/src/isolated/native_api_win32.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ffi';
+
+import 'package:ffi/ffi.dart';
+import 'package:win32/win32.dart';
+
+import '../windows/native_api.dart';
+
+/// An implementation of [NativeApi] which uses the win32 package.
+class Win32NativeApi extends NativeApi {
+  const Win32NativeApi();
+
+  @override
+  int launchApp(String amuid, List<String> args) {
```

Comment:
@clarkezone @cbracken Was an alternative design considered in which we would get this functionality from a utility bundled with the Windows engine artifacts instead of adding the `ffi` dependencies to the `flutter_tool`? For example, the Windows engine artifacts could come with a program, like `uwp_launcher` that does what is in this file, and then the `flutter_tool` could launch a uwp app using `dart:io` as usual through that launcher program?

Reply:
Having an `adb` like program for UWP would be pretty neat. I have to add that I'm indifferent between that and using ffi, but prefer both to calling out to powershell scripts. The former could give reasonable exit codes with meaning, the latter is just sort of a jumbled mess of text

Reply:
Even something less ambitious than `adb` might be easier to maintain and test than introducing `dart:ffi` to the tool for the first time. I would like to hear thoughts from @clarkezone and @cbracken on this before we land this PR because I suspect if this is landed as-is we won't take the opportunity to consider this again.

Reply:
I don't have strong opinion either way, although in the short term getting a E2E working solution is a high priority so I would prefer to get to that and improve / refactor later than do a big refactor now.